pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,179 @@
+        # embed with pre-set color of green
+
+    # our custom prompt function using buttons
+    async def prompt(
+        self,
+        message: str,
+        *,
+        timeout: float = 60.0,
+        delete_after: bool = True,
+        author_id: Optional[int] = None,
+    ) -> Optional[bool]:
+        """
+        Parameters
+        message: str
+            The message to show along with the prompt.
+
+        timeout: float (Defaults to 60 seconds)
+            How long to wait before stopping the view
+        delete_after: bool (Defaults to True)
+            Whether to delete the confirmation message after we're done.
+
+        author_id: Optional[int] (Defaults to ctx.author)
+            The member who should respond to the prompt.
+        -------
+        Returns
+            True if the confirm button is pressed,
+            False if cancel button is pressed,
+            None if automatically denied due to no response
+        """
+
+        author_id = author_id or self.author.id
+        # use the View we have made above and set the arguments
+        view = PromptView(
+            timeout=timeout,
+            delete_after=delete_after,
+            ctx=self,
+            author_id=author_id,
+        )
+
+        view.message = await self.send(message, view=view)
+        await view.wait()
+        return view.value  # Can return True, False or None
+
+
+class Bot(commands.Bot):
+    def __init__(self):
+        super().__init__(
+            # key word arguments
+        )
+
+    async def get_context(
+        self, message, *, cls=CustomContext
+    ):  # When overriding this method, you pass your new Context class to the super() method indicating that there is a new Context class the bot will use.
+        return await super().get_context(message, cls=cls)
+
+
+bot = Bot()
+
+# A command example using subclassed Context:
+@bot.command()
+async def embed(ctx):
+    await ctx.send_embed("This is a custom Context function!")
+    # This would send an embed with the description of the specified string and the embed color wold be Blurple, even though you have not specified it in making the embed, is was set as the pre-set value. You can add more kwargs in sending the embed. For example:
+
+    await ctx.send_success_embed(
+        "This is another example of a custom Context function!",
+        title="Subclassing Context",
+        timestamp=datetime.datetime.utcnow(),
+    )
+
+
+# example of the prompt function. It can be used before carrying out big or dangerous operations. For example a massban command where multiple members can be passed in to the member argument
+@bot.command()
+async def massban(ctx, members: commands.Greedy[discord.Member]):
+
+    request_confirm = await ctx.prompt(f"Are you sure you want to ban {len(members)}?")
+    if request_confirm is False:
+        # the user cancelled so we can return
+    elif request_confirm is True:
+        # the user confirmed so we can continue with the command
```

Comment:
This is invalid syntax, if you wish to have the user fill in something please add `...`